src/PactNet.Native/PactVerifier.cs

```diff
@@ -64,6 +68,36 @@ public IPactVerifier ServiceProvider(string providerName, Uri pactUri)
             return this;
         }
 
+        /// <summary>
+        /// Set the provider details
+        /// </summary>
+        /// <param name="providerName">Name of the provider</param>
+        /// <param name="pactUri">URI of the running service</param>
+        /// <param name="messagingRoute">Route where the messaging interaction is going to be sent</param>
+        /// <returns>Fluent builder</returns>
+        public IPactVerifier MessagingServiceProvider(string providerName, Uri pactUri, string messagingRoute)
+        {
+            if (string.IsNullOrWhiteSpace(providerName))
+            {
+                throw new ArgumentException("The provider name cannot be empty");
+            }
+
+            if (string.IsNullOrWhiteSpace(messagingRoute))
+            {
+                throw new ArgumentException("The messaging route cannot be null or empty");
+            }
+
+            SetProviderHost(providerName, pactUri);
+
+            verifierArgs.Add("--base-path");
+
+            var messageBasePath = pactUri.AbsolutePath != "/" ? pactUri.AbsolutePath + messagingRoute : messagingRoute;
```

Comment:
Joining URLs with `+` is probably dangerous, like if the absolute path ends with `/` and then the user specifies `/` at the start of theirs then the URL will end up `http://localhost/foo//bar` which might be break. We should join in a safer way.

Reply:
The best practice using the framework seems to use the Uri constructor.
That would be new Uri(pactUri, messagingRoute).AbsolutePath. Not sure I'm a fan but better than using Flurl and Url.Combine.
If you have a suggestion